File: El Mamoun/Lv 1/Problem 36.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

/*
 Problem 36 :Write a program to ask the user to enter:
 • Number1
 • Number 2
 • OperationType
*/

namespace Problem_36
{
    internal class Program
    {
        enum enCharType { Add='+',Sub='-',Multy='*',Div='/'};
        static enCharType ReadOperationType()
        {
            char C;

            Console.WriteLine("Pleae Enter a Operation Type?: ");
            C = Convert.ToChar(Console.ReadLine());

            return (enCharType) C;
        }
        static float ReadNumber()
        {
            float Number = 0;
            Console.WriteLine("Please Enter a Number?: ");
            Number=Convert.ToSingle(Console.ReadLine());

            return Number;
        }
        static float CalOperationType(float Number1, float Number2,enCharType OpType)
        {
            if (OpType == enCharType.Add)
                return Number1 + Number2;

            else if (OpType == enCharType.Sub)
                return Number1 - Number2;

            else if (OpType == enCharType.Div)
                return (Number1 / Number2);

            else if (OpType == enCharType.Multy)
                return (Number1 * Number2);

            else
                return 0;
        }

        static void Main(string[] args)
        {

            float Number1 = ReadNumber();
            float Number2 = ReadNumber();

            enCharType OpType = ReadOperationType();

            Console.WriteLine(CalOperationType(Number1, Number2, OpType));


            Console.ReadKey();
        }
    }
}

File: El Mamoun/Lv 1/Problem 36.cpp
#include<iostream>
using namespace std;

/*
 Problem 36 :Write a program to ask the user to enter:
 • Number1
 • Number 2
 • OperationType
*/

enum enOperationType { Add = '+', Sub = '-', Mult = '*', Div = '/' };

float ReadNumber()
{
	float number = 0;

	cout << "Please Enter a Number? ";
	cin >> number;

	return number;
}

enOperationType ReadOperationType()
{
	char Op;

	cout << "Please Enter Operation Type? ";
	cin >> Op;

	return (enOperationType) Op;
}

int CalculteOperationType(enOperationType OpType, int Number1, int Number2)
{
	if (OpType == enOperationType::Add)
		return Number1 + Number2;

	else if (OpType == enOperationType::Sub)
		return Number1 - Number2;

	else if (OpType == enOperationType::Mult)
		return Number1 * Number2;

	else if (OpType == enOperationType::Div)
		return Number1 / Number2;

	else
		return 0;
}


int main()
{

	float Number1 = ReadNumber();
	float Number2 = ReadNumber();

	enOperationType OpType = ReadOperationType();

	cout << "Result = " << CalculteOperationType(OpType, Number1, Number2) << endl;


	return 0;
}